src/ls_server.rs

```diff
@@ -251,34 +252,36 @@ impl LsService {
     }
 
     fn init(&self, id: usize, init: InitializeParams) {
-        let result = InitializeCapabilities {
+        let result = InitializeResult {
             capabilities: ServerCapabilities {
-                textDocumentSync: DocumentSyncKind::Incremental as usize,
-                hoverProvider: true,
-                completionProvider: CompletionOptions {
-                    resolveProvider: true,
-                    triggerCharacters: vec![".".to_string()],
-                },
+                text_document_sync: Some(TextDocumentSyncKind::Incremental),
```

Comment:
> the problem is when they are implicit and thus un-tested. With this pattern, they are just as explicit and thus safe to use.

That is part of it (and the most troublesome part), but I'd argue that adding an extra value just because it is possible and simplifies some parts is similar to how null pointer got so prevalent as well.